satellite/metainfo/service.go

```diff
@@ -81,10 +88,23 @@ func (s *Service) UpdatePieces(ctx context.Context, path string, ref *pb.Pointer
 
 		// put all existing pieces to a map
 		pieceMap := make(map[int32]*pb.RemotePiece)
+		nodePieceMap := make(map[storj.NodeID]*pb.RemotePiece)
 		for _, piece := range pointer.GetRemote().GetRemotePieces() {
 			pieceMap[piece.PieceNum] = piece
+			if checkDuplicates {
+				nodePieceMap[piece.NodeId] = piece
+			}
 		}
 
+		// Return an error if the pointer already has a piece for this node
+		if checkDuplicates {
+			for _, piece := range toAdd {
```

Comment:
also set `nodePieceMap[piece.NodeId] = piece` so that if someone tries to add two pieces with the same node ID, we catch it.

Reply:
nit - could nodePieceMap just be `map[storj.NodeID]struct{}{}` if we're not actually using the value?

Reply:
Nice catch